### PR TITLE
Contribute Voice Message: Add initial config for voice messages

### PR DESCRIPTION
**Pull request by JulienTant**
#### Summary
This PR adds the following elements:
- support for voice post type
- validation rules for this new post type
- configuration, including an experimental setting and a feature flag
- adds enterprise metrics support

### Has related PR

- Webapp https://github.com/mattermost/mattermost-webapp/pull/11170
- Enterprise https://github.com/mattermost/enterprise/pull/1275

#### Ticket Link
N/A

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: app/post.go [hunk 1]
```diff
@@ -369,6 +383,32 @@ func (a *App) CreatePost(c request.CTX, post *model.Post, channel *model.Channel
...
+	if err != nil {
+		return model.NewAppError("validateVoicePost", "api.post.validate_voice_post.unable_get_file_info.app_error", nil, "id="+post.Id+" fileId="+fileIds[0], http.StatusInternalServerError).Wrap(err)
+	}
+	if fileInfo == nil {
+		return model.NewAppError("validateVoicePost", "api.post.validate_voice_post.file_not_found.app_error", nil, "id="+post.Id+" fileId="+fileIds[0], http.StatusBadRequest)
+	}
+
+	if fileInfo.Size > *a.Config().FileSettings.MaxVoiceMessagesFileSize {
+		return model.NewAppError("validateVoicePost", "api.post.validate_voice_post.file_too_big.app_error", nil, "id="+post.Id+" fileId="+fileIds[0], http.StatusBadRequest)
+	}
```
Comment (streamer45) [inline app/post.go:407]: Should we add a check on `Fileinfo.MimeType` as well? We could have a constant list of formats we support (e.g. mp3/m4a).

**Review by streamer45 — COMMENTED** [review]
Looks great, just one minor suggestion to potentially fail fast in case the file type is not what we are expecting.

File: app/post.go (hunk 1 above)
Comment (JulienTant) [reply app/post.go:407]: Will add a check! Don't we support mp3 only? :confused:

Comment (streamer45) [reply app/post.go:407]: Yes, we can check for mp3 but leave a way to extend it in the future if needed.

File: model/feature_flags.go [hunk 2]
```diff
@@ -75,6 +75,8 @@ type FeatureFlags struct {
 	PlanUpgradeButtonText string
 
 	PostPriority bool
+
+	EnableVoiceMessages bool
```
Comment (M-ZubairAhmed) [inline model/feature_flags.go:79]: we have name of feature flags without prefixes. can this also be just VoiceMessage

**Review by M-ZubairAhmed — COMMENTED** [review]
Looks great

**Pull request merged**